### ISSUE TITLE
Associate marks with nodes again

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jbeder/yaml-cpp#200]
```
The new API removed a very important feature:  the ability to associate a line 
and column number with a particular element.  This is a big problem because it 
means that I have lost the ability to tell the user at what line and column an 
error occurred if they made a mistake in the input.  It would be nice if Nodes 
once again had an (optional) associated mark so that this information could be 
available.
```

Original issue reported on code.google.com by `gcrosswh...@gmail.com` on 15 Apr 2013 at 11:57

**Comment by jbeder** [comment]
Fixed by WrinklyNinja's patch, commit ec8aa4fa625257b0db7254460b91b517e2b16a7e.

**Comment by riemancon** [comment]
Can this commit be included (squashed) into the next release?  It would seem that the boost removal issue, #309, is prompting a new release and it would be great if this issue could be included.  I've tested and am using the patch with 0.5.2 and have had no problems, it's quite stable and useful as far as i can tell (I use it to report parse error locations).
